src/reducers.rs

```diff
@@ -165,6 +165,22 @@ impl Script {
         Ok(())
     }
 
+    /// Attempt to nicely tell the child to stop by sending it an empty line to
+    /// use as the next "seed", whereupon it should exit cleanly, thus cleaning
+    /// up any resources it was using (e.g. temporary files).
+    fn shutdown_child(&mut self) {
+        if let Some(mut child) = self.child.take() {
+            if {
+                let mut child_stdin = child.stdin.as_mut().unwrap();
+                write!(child_stdin, "\n").is_err()
```

Comment:
I think we should also close stdin after writing the newline and wait on the child here (and `wait` will close stdin for us,s o two birds with one stone):

```rust
write!(child_stdin, "\n").and_then(|_| self.child.wait()).is_err()
```